client/src/main/java/org/apache/rocketmq/schema/registry/client/CachedSchemaRegistryClient.java

```diff
@@ -0,0 +1,485 @@
+        if (subjectToVersion.get(subjectFullName) != null) {
+            subjectToVersion.get(subjectFullName).forEach(
+                version -> schemaCacheBySubjectAndVersion.invalidate(new SubjectAndVersion(cluster, tenant, subject, version)));
+        }
+        //invalidate schemaCacheBySubjectAndId
+        if (subjectToId.get(subjectFullName) != null) {
+            subjectToId.get(subjectFullName).forEach(
+                recordId -> schemaCacheBySubjectAndId.invalidate(new SubjectAndId(cluster, tenant, subject, recordId)));
+        }
+        // invalidate schemaCache
+        if (subjectToSchema.get(subjectFullName) != null) {
+            subjectToSchema.get(subjectFullName).forEach(
+                schema -> schemaCache.invalidate(new SubjectAndSchema(cluster, tenant, subject, schema)));
+        }
+        subjectToVersion.remove(subjectFullName);
+        subjectToId.remove(subjectFullName);
+        subjectToSchema.remove(subjectFullName);
+        return restService.deleteSchema(cluster, tenant, subject);
+    }
+
+    @Override
+    public DeleteSchemeResponse deleteSchema(String cluster, String tenant, String subject,
+        long version) throws IOException, RestClientException {
+        String subjectFullName = String.format("%s/%s/%s", cluster, tenant, subject);
+        schemaCacheBySubject.invalidate(subjectFullName);
+        schemaCacheBySubjectAndVersion.invalidate(new SubjectAndVersion(cluster, tenant, subject, version));
+        subjectToVersion.get(subjectFullName).remove(version);
+
+        return restService.deleteSchema(cluster, tenant, subject, version);
+    }
+
+    @Override
+    public UpdateSchemaResponse updateSchema(String subject, String schemaName,
+        UpdateSchemaRequest request) throws RestClientException, IOException {
+        // invalidate schemaCache
+        schemaCache.invalidate(new SubjectAndSchema(DEFAULT_CLUSTER, DEFAULT_TENANT, subject, schemaName));
+        return restService.updateSchema(subject, schemaName, request);
+    }
+
+    @Override
+    public UpdateSchemaResponse updateSchema(String cluster, String tenant, String subjectName,
+        String schemaName, UpdateSchemaRequest request) throws IOException, RestClientException {
+        // invalidate schemaCache
+        schemaCache.invalidate(new SubjectAndSchema(cluster, tenant, subjectName, schemaName));
+        return restService.updateSchema(cluster, tenant, subjectName, schemaName, request);
+    }
+
+    @Override
+    public GetSchemaResponse getSchemaBySubject(String subject) throws RestClientException, IOException {
+        String fullName = String.format("%s/%s/%s", DEFAULT_CLUSTER, DEFAULT_TENANT, subject);
+        GetSchemaResponse result = schemaCacheBySubject.getIfPresent(fullName);
+        if (result != null) {
+            return result;
+        }
+        result = restService.getSchemaBySubject(subject);
+        schemaCacheBySubject.put(fullName, result);
```

Comment:
Maybe cache update mechanism can be write like:

`
        Cache<Integer, String> a = CacheBuilder.newBuilder().build(
            new CacheLoader<Integer, String>() {
                @Override
                public String load(Integer key) throws Exception {
                    return restService.getSchemaBySubject(subject);
                }
            }
`

Reply:
I'm sorry, I didn't understand this. Is this code used in the initializing process? Could you show me how to use this code